### PR TITLE
change slice declarations to array where length already obvious

### PR DESCRIPTION
**Pull request by mstrYoda**
# Description

I change some slice declarations to array when the length is already known while initialization.
It might be a tiny performance improvement by preventing dynamic sizing.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #_[issue number]_

#2955 https://github.com/dapr/dapr/issues/2955

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [X] Code compiles correctly
* [ ] Created/updated tests
* [X] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Review by pkedy — COMMENTED** [review]
@mstrYoda  If you fix the one `gofmt issue` then it LGTM.

**Comment by mstrYoda** [comment]
> @mstrYoda If you fix the one `gofmt issue` then it LGTM.

Thank you for your review, I pushed a commit with go fmt 👍

File: pkg/http/api.go [hunk 1]
```diff
@@ -1134,7 +1134,7 @@ func (a *api) onGetMetadata(reqCtx *fasthttp.RequestCtx) {
 		activeActorsCount = a.actor.GetActiveActorsCount(reqCtx)
 	}
 
-	registeredComponents := []registeredComponent{}
+	registeredComponents := make([]registeredComponent, 0, len(a.getComponentsFn()))
```
Comment (luckyxiaoqiang) [inline pkg/http/api.go:1137]: We can call `a.getComponentsFn()` only once, because there are slice allocation and copy every time it is called.

Comment (mstrYoda) [reply pkg/http/api.go]: Sorry, my mistake. I changed that line. Calling len using components variable.

**Review by luckyxiaoqiang — APPROVED** [review]

**Review by pkedy — APPROVED** [review]

**Pull request merged**